dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tool.java

```diff
@@ -382,7 +384,7 @@ public Registry getRegistryProvider() {
         }
 
         // Deal with registries with custom registry paths
-        if (this.registry.matches("^[a-zA-Z0-9]+\\.dkr\\.ecr\\.[a-zA-Z0-9]+\\.amazonaws\\.com")) {
+        if (this.registry.matches(AMAZON_ECR_PRIVATE_REGISTRY_REGEX)) {
```

Comment:
This probably isn't called frequently enough to matter, and it was already like this, so you can leave it, but it's generally a [best practice](https://stackoverflow.com/questions/1720191/java-util-regex-importance-of-pattern-compile) to compile regex's just once (`string.matches` invokes `Pattern.compile`).